arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -893,19 +893,13 @@ public fun <A, K, V> Iterable<A>.crosswalkMap(f: (A) -> Map<K, V>): Map<K, List<
       ior.fold(
         { listOf(it) },
         ::identity,
-        { l, r -> listOf(l) + r }
```

Comment:
I really would prefer not using `fold` here at all, and instead using `buildMap` and an imperative implementation. I understand that this is likely a standard implementation though, so I get why it's written this way. Maybe we can change it in another PR I guess